src/script/lua_api/l_base.cpp

```diff
@@ -90,3 +90,15 @@ bool ModApiBase::isNaN(lua_State *L, int idx)
 {
 	return lua_type(L, idx) == LUA_TNUMBER && std::isnan(lua_tonumber(L, idx));
 }
+
+/*
+ * Read template functions
+ */
+template<>
+float ModApiBase::read(lua_State *L, int index)
+{
+	if (isNaN(L, index))
+		throw LuaError("NaN value is not allowed.");
+
+	return (float) luaL_checknumber(L, index);
```

Comment:
`read<T>` implies that multiple types can be read, like bool and integer. Yet it only supports floats, which lose are not accurate enough for 32-bit integers or higher.

Reply:
using template function permits to prevent doing many luaReadXXX things and auto generate the function,s it's same, but it's  cleaner c+++

Reply:
currently yes we only have read<float> but i will introduce read<int>, read<u8>, read<double> and more :) this will permit to securize our Lua API calls a little bit more with a common high level interface.
also i will migrate some common tools like read<v3f> :)